### PR TITLE
Created backend files for DM system, added frontend formatting

### PR DESCRIPTION
**Pull request by gretaLiesch**
Created backend methods for getting information from database to add to frontend layout (eg. current DM chats). Added appropriate frontend java functionality to display the obtained information. Fixed references to DM system filenames in other parts of the program.

*NOTE: backend is not fully functional yet, so these files cannot be run as they are now, I'm just making a pull request for completeness*

### CODE REVIEW

File: WebContent/dmpageback.jsp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                start = i + 1;
+                count += 1;
+            }
+        }
+        else {
+            if (thread.charAt(i) == '`') {
+                end = i - 1;
+
+                // Add userID and message to arraylist
+                messages.add(userId, thread[start:end]);
```
Comment (aaan-gelina) [inline WebContent/dmpageback.jsp:80]: Error in original funciton, must be corrected the format as a pair object: messages.add(new Pair<Integer, String>(userId, thread[start:end]));

**Review by aaan-gelina — CHANGES_REQUESTED** [review]
Please see comment and update code to merge branch!

File: WebContent/dmpageback.jsp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    // Add new message to thread
+    thread = thread.append(newMessage);
+
+    // Create database reference
+    DatabaseReference root = FirebaseDatabase.getInstance().getReference();
+    DatabaseReference currentChannel = root.child("Channels").child(cid);
+
+    //Set Messages value in database to update thread
+    currentChannel.child(messages).setValue(thread);
```
Comment (aaan-gelina) [inline WebContent/dmpageback.jsp:112]: add quotes around messages -> "messages"

**Review by aaan-gelina — CHANGES_REQUESTED** [review]
Found a bug in database call, need to add quotes when referencing a db key

File: WebContent/dmpageback.jsp (hunk 1 above)
Comment (gretaLiesch) [reply WebContent/dmpageback.jsp]: Fixed it:)

File: WebContent/dmpageback.jsp (hunk 2 above)
Comment (gretaLiesch) [reply WebContent/dmpageback.jsp]: Fixed as well.

**Review by aaan-gelina — APPROVED** [review]
Looks good with the changes!

**Pull request merged**